tokio/src/io/mod.rs

```diff
@@ -15,19 +15,19 @@
 //! type will _yield_ to the Tokio scheduler when IO is not ready, rather than
 //! blocking. This allows other tasks to run while waiting on IO.
 //!
-//! Another difference is that [`AsyncRead`] and [`AsyncWrite`] only contain
+//! Another difference is that `AsyncRead` and `AsyncWrite` only contain
 //! core methods needed to provide asynchronous reading and writing
 //! functionality. Instead, utility methods are defined in the [`AsyncReadExt`]
 //! and [`AsyncWriteExt`] extension traits. These traits are automatically
-//! implemented for all values that implement [`AsyncRead`] and [`AsyncWrite`]
+//! implemented for all values that implement `AsyncRead` and `AsyncWrite`
 //! respectively.
 //!
-//! End users will rarely interact directly with [`AsyncRead`] and
-//! [`AsyncWrite`]. Instead, they will use the async functions defined in the
-//! extension traits. Library authors are expected to implement [`AsyncRead`]
-//! and [`AsyncWrite`] in order to provide types that behave like byte streams.
+//! End users will rarely interact directly with `AsyncRead` and
+//! `AsyncWrite`. Instead, they will use the async functions defined in the
+//! extension traits. Library authors are expected to implement `AsyncRead`
+//! and `AsyncWrite` in order to provide types that behave like byte streams.
 //!
-//! Even with these differences, Tokio's [`AsyncRead`] and [`AsyncWrite`] traits
+//! Even with these differences, Tokio's `AsyncRead` and `AsyncWrite` traits
```

Comment:
You're removing a lot of links here. What is the motivation for that?

Reply:
Definitely an opinionated change.  My thought is that while having types be links to their type page is super helpful, in this section `AsyncRead` and `AsyncWrite` appear so many times that the number of links is a bit large.

What I did was to change the links so that each type is linked the first time it is mentioned.  This has the effect of displaying each unique type mentioned in this paragraph as links, with the different formatting that implies.

Personally, I think the benefit to being able to "at a glance" see most of the types that are associated with `AsyncRead` and `AsyncWrite` outweights having to move your mouse a little farther.  I also tried to make it so that about every screenful of text had at least one link to the type if it was mentioned, because I think it's important to avoid the minor frustration of not having a link for a type you're reading about.

Sound reasonable?

Reply:
That might make sense, as long as the link is nearby and easy to find. The documentation doesn't compile right now, so I couldn't check out how it looks without the links, but once the links are fixed, I'll take a look. :+1: